---
- debug: msg="START cli/invalid_subset.yaml on connection={{ ansible_connection }}"

- name: test invalid subset (foobar)
  eos_facts:
    gather_subset:
      - "foobar"
    provider: "{{ cli }}"
  become: yes
  register: result
  ignore_errors: true


- assert:
    that:
      # Failures shouldn't return changes
      - "result.changed == false"
      # It's a failure
      - "result.failed == true"
      # Sensible Failure message
      - "'Subset must be one of' in result.msg"

- debug: msg="END cli/invalid_subset.yaml on connection={{ ansible_connection }}"
